### PR TITLE
Add vec256_half_neon

### PR DESCRIPTION
**Pull request by metascroy**
Summary: Add `vec256_half_neon.h` Currently not used anywhere.

Differential Revision: D55429392

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Review by mikekgfb — APPROVED** [review]
LGTM

File: aten/src/ATen/cpu/vec/vec256/vec256_half_neon.h [hunk 1]
```diff
@@ -0,0 +1,828 @@
...
+
+#ifdef __BIG_ENDIAN__
+#error "Big endian is not supported."
+#endif
+
+#if defined(AT_BUILD_ARM_VEC256_WITH_SLEEF)
+#define USE_SLEEF(sleef_code, non_sleef_code) sleef_code
+#else
+#define USE_SLEEF(sleef_code, non_sleef_code) non_sleef_code
+#endif
```
Comment (malfet) [inline aten/src/ATen/cpu/vec/vec256/vec256_half_neon.h:36]: Not needed, is it?
```suggestion
```

Comment (metascroy) [reply aten/src/ATen/cpu/vec/vec256/vec256_half_neon.h:36]: Good catch.  I removed it in [D55621933](https://www.internalfb.com/diff/D55621933)

**Pull request closed without merging**